packages/wrangler/src/api/pages/deploy.tsx

```diff
@@ -212,7 +212,7 @@ export async function deploy({
 	}
 
 	if (commitMessage) {
-		formData.append("commit_message", commitMessage);
+		formData.append("commit_message", commitMessage.replaceAll('\\n', '\n'));
```

Comment:
it appears that someone put this line here on purpose, but I can't think of why this would be desirable. Anyone knows?

Reply:
what do you mean? the `replaceAll` I'm adding?

anyways sorry please ignore it as I need to improve the solution (as it should be clever enough not to unescape newlines that the users are trying to escape)

Reply:
hey @dario-piotrowicz, apologies I misread this diff as you removing the line, which is incorrect.

But the bigger question in the background stands, do we know how this newline gets double escaped? What causes that to happen?

Reply:
no, I am not sure what's causing that, I just know that `wrangler pages deploy --commit-message` does that (so it's not a C3 related thing) I haven't figured out yet what's causing that specifically I'll investigate further 🙇